game_patch/multi/multi.cpp

```diff
@@ -245,14 +246,27 @@ bool is_entity_out_of_ammo(rf::Entity *entity, int weapon_type, bool alt_fire)
     return clip_ammo == 0;
 }
 
-std::pair<bool, float> multi_is_cps_above_limit(rf::Player* pp, float max_cps)
+std::pair<bool, float> multi_is_cps_above_limit(rf::Player* pp, float max_cps, int weapon_type, bool alt_fire)
 {
     constexpr int num_samples = 4;
     int player_id = pp->net_data->player_id;
+    static int last_weapon_id[rf::multi_max_player_id];         // Keep a record of last weapon used
+    static bool last_weapon_fire_mode[rf::multi_max_player_id]; // Keep a record of last fire mode used
     static int last_weapon_fire[rf::multi_max_player_id][num_samples];
     int now = rf::timer_get(1000);
+    // If the weapon or fire mode has changed, zero out the saved array of timestamps. Avoids dropping legit fire packets based on cps from other weapon/firemode
+    if (last_weapon_id[player_id] != weapon_type || last_weapon_fire_mode[player_id] != alt_fire) {
```

Comment:
is this really a bug? I left those checks intentionally out because switching weapon is long enough process compared to CPS values we are talking about here, that it shouldn't matter. Can you give an example when it caused problems? Is it the riot stick?